### PR TITLE
hotfix - Fix dataone URI harvesting

### PR DESCRIPTION
**Pull request by erinspace**
- Add use_doi kwarg option to standard URI processor 
- Make the dataone harvester use this kwarg and use the DOIs for the canonicalUri

### CODE REVIEW

File: scrapi/base/helpers.py [hunk 1]
```diff
@@ -135,7 +135,10 @@ def oai_process_uris(*args):
                 provider_uris.append(found_url)
 
     try:
-        canonical_uri = (provider_uris + object_uris)[0]
+        if kwargs.get('use_doi') and len(object_uris) > 0:
```
Comment (fabianvf) [inline scrapi/base/helpers.py]: Would be better to make this a top-level named argument, rather than put it in kwargs.

**Comment by erinspace** [comment]
Also, misnamed my git branch - it is indeed a hotfix branch though!

File: scrapi/base/helpers.py [hunk 2]
```diff
@@ -135,7 +136,10 @@ def oai_process_uris(*args):
                 provider_uris.append(found_url)
 
     try:
-        canonical_uri = (provider_uris + object_uris)[0]
+        if use_doi and len(object_uris) > 0:
+            canonical_uri = object_uris[0]
```
Comment (fabianvf) [inline scrapi/base/helpers.py]: Are object_uris guaranteed to be DOIs?

Comment (fabianvf) [inline scrapi/base/helpers.py]: Also, do harvesters that use this flag not have any providerUris? The canonicalUri would ideally be some sort of splash page

File: scrapi/base/helpers.py [hunk 3]
```diff
@@ -135,7 +136,10 @@ def oai_process_uris(*args):
...
+        else:
+            canonical_uri = (provider_uris + object_uris)[0].strip().replace(']', '')
```
Comment (fabianvf) [inline scrapi/base/helpers.py]: Is this `.replace` still necessary, or did the regex fix that issue?

File: scrapi/base/helpers.py (hunk 2 above)
Comment (erinspace) [inline scrapi/base/helpers.py]: A doi oftentimes does point to the sources splash page where the resource lives -- it does on dataone and others I've seen for sure

Comment (fabianvf) [inline scrapi/base/helpers.py]: hmm, I guess there is no way to split out DOIs pointing to splash pages vs elsewhere. Though if DOIs usually point to splash pages, should they be in providerUris instead?

**Comment by erinspace** [comment]
Incorporated all these changes into #372, closing this PR!

**Pull request closed without merging**